mapper/env.go

```diff
@@ -132,9 +133,12 @@ func (e EnvironmentMapper) ToInternal(data map[string]interface{}) {
 				})
 			}
 		} else {
+			if targetKey == "" {
+				targetKey = sourceKey
+			}
 			switch source {
 			case "field":
-				envVar = append(envVarFrom, map[string]interface{}{
+				envVar = append(envVar, map[string]interface{}{
```

Comment:
This is a separate bug - need to append to `envVar` not `envVarFrom`.

Reply:
@kinarashah can we link this bug to PR as well? and if not there, we should file it so qa can verify it